lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
+  switch (space) {
+    case 'hsl':
+    case 'hwb':
+    case 'lch':
+    case 'oklch':
+      return true;
+    default:
+      return false;
+  }
+}
+
+/**
+ * Normalize between ColorJS coordinates (which use `NaN`) and Sass Color
+ * coordinates (which use `null`).
+ */
+function getCoordsFromColor(
+  coords: [number, number, number],
+  isRgb = false
+): [number | null, number | null, number | null] {
+  let newCoords: [number | null, number | null, number | null] = coords;
+  if (isRgb) {
+    newCoords = (newCoords as [number, number, number]).map(coordToRgb) as [
+      number,
+      number,
+      number,
+    ];
+  }
+  return (newCoords as [number, number, number]).map(NaNtoNull) as [
+    number | null,
+    number | null,
+    number | null,
+  ];
+}
+
+function propertyIsSet(val: undefined | null | number): val is number | null {
+  return val === null || typeof val === 'number';
+}
+
+function checkChangeDeprecations(
+  options: {
+    [key in ChannelName]?: number | null;
+  },
+  channels: ChannelName[]
+) {
+  if (options.alpha === null) {
+    emitNullAlphaDeprecation();
+  }
+  for (const channel of channels) {
+    if (options[channel] === null) {
+      emitColor4ApiChangeNullDeprecation(channel);
+    }
+  }
+}
+
+function emitColor4ApiGetterDeprecation(name: string) {
+  console.warn(
+    `Deprecation [color-4-api]: \`${name}\` is deprecated; use \`channel\` instead.`
+  );
 }
 
-interface HslColor {
-  hue: number;
-  saturation: number;
-  lightness: number;
-  alpha?: number;
+function emitColor4ApiChangeSpaceDeprecation() {
+  console.warn(
+    'Deprecation [color-4-api]: ' +
+      "Changing a channel not in this color's space without explicitly " +
+      'specifying the `space` option is deprecated.'
+  );
 }
 
-interface HwbColor {
-  hue: number;
-  whiteness: number;
-  blackness: number;
-  alpha?: number;
+function emitColor4ApiChangeNullDeprecation(channel: string) {
+  console.warn(
+    'Deprecation [color-4-api]: ' +
+      `Passing \`${channel}: null\` without setting \`space\` is deprecated.`
+  );
+}
+
+function emitNullAlphaDeprecation() {
+  console.warn(
+    'Deprecation [null-alpha]: ' +
+      'Passing `alpha: null` without setting `space` is deprecated.' +
+      '\n' +
+      'More info: https://sass-lang.com/d/null-alpha'
+  );
 }
 
 /** A SassScript color. */
 export class SassColor extends Value {
-  private redInternal?: number;
-  private greenInternal?: number;
-  private blueInternal?: number;
-  private hueInternal?: number;
-  private saturationInternal?: number;
-  private lightnessInternal?: number;
-  private readonly alphaInternal: number;
-
-  constructor(color: RgbColor);
-  constructor(color: HslColor);
-  constructor(color: HwbColor);
-  constructor(color: RgbColor | HslColor | HwbColor) {
+  // ColorJS color object
+  private color: ColorType;
+  // Boolean indicating whether this color is in RGB format
```

Comment:
I think this warrants a bit more explanation about the difference between the RGB and sRGB color spaces.